### PR TITLE
ASTRO-527 - Replace Bitbucket links | Demo Update on UXDS

### PR DESCRIPTION
**Pull request by im-cr**
TASK
https://rocketcom.atlassian.net/browse/ASTRO-527

- Updat the component links and template linking AUXDS component element. 

IMPACTED AREAS

- AUXDS component pages. Specifically, component links to github.com and related component directories.

### CODE REVIEW

File: _content/components/checkbox.md [hunk 1]
```diff
@@ -6,7 +6,6 @@ layout: components.template.njk
 title: Checkbox
 demo: https://rocketcom.bitbucket.io/html-demos/checkbox.html
```
Comment (sknep) [inline _content/components/checkbox.md:7]: The "demo" property still links to bitbucket stuff? Is that intentional/correct?

File: _includes/components.template.njk [hunk 2]
```diff
@@ -20,7 +20,7 @@
 					</a>
 				{% endif %}
 				{% if git %}
-					<a class="sample-links__bitbucket" href="https://bitbucket.org/rocketcom/astro-components/src/master/src/components/{{git}}">Bitbucket</a>
+					<a class="sample-links__github" href="https://github.com/RocketCommunicationsInc/astro-components/tree/master/src/components/{{git}}">Github</a>
```
Comment (sknep) [inline _includes/components.template.njk:23]: What happens if the component markdown file doesn't have a `git` property? The checkbox file above has it removed.

File: _content/components/checkbox.md [hunk 3]
```diff
@@ -6,7 +6,6 @@ layout: components.template.njk
...
 storybook: components-form-elements--checkboxes
-git: rux-checkbox
```
Comment (sknep) [inline _content/components/checkbox.md:9]: Clicking the github link from this page https://deploy-preview-69--astrouxds.netlify.app/components/checkbox/ results in a 404 https://github.com/RocketCommunicationsInc/astro-components/tree/master/src/components/[object%20Object]

Input field page also 404s with the same URL (missing `git`  YAML property)

Notification Banner has the wrong `git` value, which creates a URL that 404s: https://github.com/RocketCommunicationsInc/astro-components/tree/master/src/components/rux-banner

Same with Radio Button: https://github.com/RocketCommunicationsInc/astro-components/tree/master/src/components/rux-radio-button

and popup menu: https://github.com/RocketCommunicationsInc/astro-components/tree/master/src/components/pop-up-menu

There are probably more, I stopped checking here.

**Comment by sknep** [comment]
I can't tell from the Exit Criteria on the ticket, but should the Github links be linking to an example of the component or to the component source itself? I thought the ticket had to do with demos, not source code.

**Comment by sknep** [comment]
Can we add a little bit about the task to the name of the PR so that later we know what it was about without having to visit Jira? Maybe something like "Fix Github links in component demo containers"?

**Comment by dmcalester** [comment]
> I can't tell from the Exit Criteria on the ticket, but should the Github links be linking to an example of the component or to the component source itself? I thought the ticket had to do with demos, not source code.

Component Source.

File: _content/components/checkbox.md (hunk 1 above)
Comment (dmcalester) [reply _content/components/checkbox.md:7]: @im-cr see if you can update the demo link to refer to this page https://astro-components.netlify.app/?path=/story/components-form-elements--checkboxes

Following the storybook demo convention seen on other pages.

File: _includes/components.template.njk (hunk 2 above)
Comment (dmcalester) [reply _includes/components.template.njk:23]: Won't show. The form elements don’t have git pages as they’re bare elements

File: _content/components/checkbox.md (hunk 1 above)
Comment (sknep) [reply _content/components/checkbox.md:7]: input fields, too?

Comment (dmcalester) [reply _content/components/checkbox.md:7]: @sknep  The input fields one on Storybook is huge. I wouldn’t be opposed to another story that could be used, but current isn’t practical

File: _includes/components.template.njk (hunk 2 above)
Comment (sknep) [reply _includes/components.template.njk:23]: https://deploy-preview-69--astrouxds.netlify.app/components/checkbox/
https://deploy-preview-69--astrouxds.netlify.app/components/input-field/ 
Are you sure? I'm thinking the git conditional isn't working the way we think it is?

Comment (dmcalester) [reply _includes/components.template.njk:23]: @sknep good catch, that isn’t working properly. @im-cr please update to have the GitHub link only show when there is a `git` property set.

Comment (im-cr) [reply _includes/components.template.njk:23]: Caught this while push last update, conditional output is still generating that markup on site build, working through solution.

Comment (im-cr) [reply _includes/components.template.njk:23]: resolved

File: _content/components/checkbox.md (hunk 3 above)
Comment (im-cr) [reply _content/components/checkbox.md:9]: @dmcalester @sknep Updated github and storybook links. Any components that do not have a corresponding github component have been removed. Story demo links have been updated as well. Some have been left in the previous story linking format due to content length of individual component (i.e button variants - https://astro-components.netlify.app/?path=/story/components-buttons--all-button-variants).

In cases where these is no appropriate demo or component dir ('search') I've removed both.

**Comment by dmcalester** [comment]
Thanks @im-cr 

Couple of notes

[Status Symbols](https://deploy-preview-69--astrouxds.netlify.app/components/status-symbol/) should have a GitHub link (https://github.com/RocketCommunicationsInc/astro-components/tree/master/src/components/rux-status)

Overall the presentation of the components is a bit hit and miss. Some have a lot of padding around them, some have very little at all. This is covered in another ticket, just want to make sure it isn’t lost https://rocketcom.atlassian.net/browse/ASTRO-533

**Comment by im-cr** [comment]
> Thanks @im-cr
> 
> Couple of notes
> 
> [Status Symbols](https://deploy-preview-69--astrouxds.netlify.app/components/status-symbol/) should have a GitHub link (https://github.com/RocketCommunicationsInc/astro-components/tree/master/src/components/rux-status)
> 
> Overall the presentation of the components is a bit hit and miss. Some have a lot of padding around them, some have very little at all. This is covered in another ticket, just want to make sure it isn’t lost https://rocketcom.atlassian.net/browse/ASTRO-533

Thanks @dmcalester 

Updated status github link and vertical padding below the demo area.  In regards to the demo container, is there any concern around the inline padding/height/spacing of the demo templates being embedded? For instance, the checkox demo has inline styling that uses a larger top padding than the other demo components. 

I am currently adjusting heights to have more consistency across demos but there will still be some parody around component padding/styles due to any inline styles from the story component demo iframes.

**Comment by im-cr** [comment]
@dmcalester Updated notes on [Astro 533](https://rocketcom.atlassian.net/browse/ASTRO-533). Demo updates in PR for [Astro533](https://github.com/RocketCommunicationsInc/astro-components/pull/13) New demo urls in place for toggle and segmented buttons for better spacing and height adjusted on tree component demo.

File: scss/sample.scss [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+	padding: 1rem;
+	border-radius: 5px;
+	background-color: var(--demo-container-background-color, #1a2634);
+  
+	iframe {
+	  box-sizing: border-box;
+	  width: 100%;
+	  padding: 0rem;
+
+	  #root-inner {
```
Comment (sknep) [inline scss/sample.scss:13]: I don't think you meant to keep these styles -- they wouldn't apply within the iFrame anyways?

File: scss/sample.scss [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+  
+	  &:not(:last-of-type) {
+		margin-right: 1rem;
+	  }
+	}
+  }
+  
+  .sample-links__storybook::before {
+	margin-right: 0.375rem;
+	background-image: url(/img/_site/storybook-logo.svg);
```
Comment (sknep) [inline scss/sample.scss:56]: I don't remember if we have a code style preference around this, but I always feel better when strings are quoted, even if this works.

File: scss/sample.scss [hunk 6]
```diff
@@ -0,0 +1,163 @@
...
+  
+	span {
+	  display: flex;
+	  align-content: center;
+	  justify-content: center;
+	  align-items: center;
+	  width: 50%;
+	  height: 100%;
+	  z-index: 10;
+	  transition: color ease-out 0.167s;
```
Comment (sknep) [inline scss/sample.scss:129]: I love a nice subtle transition!

File: scss/sample.scss [hunk 7]
```diff
@@ -0,0 +1,163 @@
...
+  }
+  
+  .sample-links__storybook::before {
+	margin-right: 0.375rem;
+	background-image: url(/img/_site/storybook-logo.svg);
+  }
+  
+  .sample-links__bitbucket::before {
+	margin-right: 0.5rem;
+	background-image: url(/img/_site/bitbucket-logo.svg);
```
Comment (sknep) [inline scss/sample.scss:61]: Do we need bitbucket anymore?

File: _content/components/spectrum-analyzer.md [hunk 8]
```diff
@@ -5,6 +5,7 @@ date: Last Modified
 layout: components.template.njk
 title: Spectrum Analyzer
 demo: https://rocketcom.bitbucket.io/packages/rux-spectrum-analyzer/demo/rux-spectrum-analyzer.html
```
Comment (sknep) [inline _content/components/spectrum-analyzer.md:7]: Is this one still on bitbucket?

File: _content/components/checkbox.md (hunk 3 above)
Comment (sknep) [reply _content/components/checkbox.md:9]: I saw some bitbucket demo urls are still in a few of the YAML files:
`demo: https://rocketcom.bitbucket.io/html-demos/**`

If we aren't using them, maybe we could remove those? there's only a few left.

File: _content/components/spectrum-analyzer.md (hunk 8 above)
Comment (dmcalester) [reply _content/components/spectrum-analyzer.md:7]: Yes. It should be deprecated in the next release.

**Comment by im-cr** [comment]
@dmcalester @sknep unecessary removed css and changed a few more urls to remove bitbucket reference. Astro-506 is a ticket related to the removal of all bitbucket references, I'll leave the remaining updates for that ticket since this one was just the removal of the icon link.

**Comment by dmcalester** [comment]
@im-cr approved to merge to master at your earliest convenience

**Comment by dmcalester** [comment]
Clicked the wrong button sorry @im-cr . Please merge and close this branch at your earliest convenience.

**Pull request merged**